### PR TITLE
fix: LegacySettingsMigrations

### PR DESCRIPTION
**Pull request by IceToast**
This PR fixes the settingsmigration for all old settings as described here:
https://discord.com/channels/787829714483019826/851144461361872957/1203989679557312552

image

### CODE REVIEW

File: app/Classes/LegacySettingsMigration.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public function getOldValue(string $key,  int|string|bool|null $default = null)
+    {
+        $old_value = DB::table('settings_old')->where('key', '=', $key)->get(['value', 'type'])->first();
+
+        if (is_null($old_value) || is_null($old_value->value)) {
+            return $default;
+        }
+
+        switch ($old_value->type) {
+            case 'string':
```
Comment (Ferks-FK) [inline app/Classes/LegacySettingsMigration.php:48]: Don't need a return for the `string` type?

**Review by Ferks-FK — CHANGES_REQUESTED** [review]
Just a few changes.

File: database/settings/2023_02_01_181334_create_pterodactyl_settings.php [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-        $this->migrator->add('pterodactyl.admin_token', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:TOKEN') : env('PTERODACTYL_TOKEN', ''));
-        $this->migrator->add('pterodactyl.user_token', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:ADMIN_USER_TOKEN') : '');
-        $this->migrator->add('pterodactyl.panel_url', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:URL') : env('PTERODACTYL_URL', ''));
-        $this->migrator->add('pterodactyl.per_page_limit', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:PER_PAGE_LIMIT') : 200);
+        $this->migrator->addEncrypted('pterodactyl.admin_token', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:TOKEN', '') : env('PTERODACTYL_TOKEN', ''));
+        $this->migrator->addEncrypted('pterodactyl.user_token', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:ADMIN_USER_TOKEN', '') : '');
+        // $this->migrator->add('pterodactyl.admin_token', $table_exists ? $this->getOldValue('SETTINGS::SYSTEM:PTERODACTYL:TOKEN') : env('PTERODACTYL_TOKEN', ''));
```
[the remark this answers is not in the document]
Comment (IceToast) [reply database/settings/2023_02_01_181334_create_pterodactyl_settings.php:14]: True that!

File: routes/web.php [hunk 3]
```diff
@@ -77,11 +77,13 @@
     Route::patch('/servers/cancel/{server}', [ServerController::class, 'cancel'])->name('servers.cancel');
     Route::resource('servers', ServerController::class);
 
-    if (config('app.key')) {
+    try {
         $serverSettings = app(App\Settings\ServerSettings::class);
-        if ($serverSettings->enable_upgrade) {
-            Route::post('servers/{server}/upgrade', [ServerController::class, 'upgrade'])->name('servers.upgrade');
+        if ($serverSettings->creation_enabled) {
```
[the remark this answers is not in the document]
Comment (1day2die) [reply routes/web.php:82]: rest lgtm

**Comment by 1day2die** [comment]
lgtm

**Pull request merged**